### PR TITLE
faker-cxx: add recipe

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **faker-cxx/***

- Version 2.0.0 has been released (https://github.com/cieslarmichal/faker-cxx/releases/tag/v2.0.0)

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by toge** [comment]
@RubenRBS 
> c++: fatal error: Killed signal terminated program cc1plus

Since faker-cxx has several large source files, CCI seems to kill the compiler process.
Is there any way to resolve this?

**Comment by uilianries** [comment]
@toge We could add a larger container to this project. Let me check it.

**Comment by meftunca** [comment]
Any progress on the project? @uilianries

File: recipes/faker-cxx/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        deps = CMakeDeps(self)
+        deps.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, pattern="LICENSE", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
```
Comment (uilianries) [inline recipes/faker-cxx/all/conanfile.py:91]:
```suggestion
        copy(self, pattern="LICENSE*", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
```

There are more licenses associated, they from parts of the project that is using from external sources.

**Comment by uilianries** [comment]
> ### Conan v1 pipeline ✔️
> 
> All green in build 9 (`1114251fdadc1f346ac3e5042ef2a1716bd19ce6`):
> 
>     * **faker-cxx/2.0.0:**
>       All packages built successfully! ([All logs](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/24018/9/faker-cxx/2.0.0//summary.json))
> 
> 
> ### Conan v2 pipeline ❌
> 
> > **Note**: Conan v2 builds are now mandatory. Please read our [discussion](https://github.com/conan-io/conan-center-index/discussions/19104) about it.
> 
> The v2 pipeline failed. Please, review the errors and note this is required for pull requests to be merged. In case this recipe is still not ported to Conan 2.x, please, ping `@conan-io/barbarians` on the PR and we will help you.
> 
> Failure in build 9 (`1114251fdadc1f346ac3e5042ef2a1716bd19ce6`):
> 
>     * **faker-cxx/2.0.0:**
>       CI failed to create some packages ([All logs](https://c3i.jfrog.io/c3i/misc-v2/summary.html?json=https://c3i.jfrog.io/c3i/misc-v2/logs/pr/24018/9-windows-msvc/faker-cxx/2.0.0//summary.json))
>       Logs for packageID `01c9f3604dd77253527cc9af19c3955254c76cc6`:
>       ```
>       [settings]
>       arch=x86_64
>       build_type=Release
>       compiler=msvc
>       compiler.cppstd=20
>       compiler.runtime=dynamic
>       compiler.runtime_type=Release
>       compiler.version=193
>       os=Windows
>       [options]
>       */*:shared=True
>       ```
>       
>       
>           
>             
>           
>       
>             
>           
>       
>           
>         
>       ```
>       [...]
>       WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
>       WARN: deprecated:     'cpp_info.names' used in: fmt/10.2.1
>       
>       ======== Testing the package ========
>       Removing previously existing 'test_package' build folder: C:\J\workspace\prod-v2\bsr\cci-4a790a97\recipes\faker-cxx\all\test_package\build\msvc-193-x86_64-20-release
>       faker-cxx/2.0.0 (test package): Test package build: build\msvc-193-x86_64-20-release
>       faker-cxx/2.0.0 (test package): Test package build folder: C:\J\workspace\prod-v2\bsr\cci-4a790a97\recipes\faker-cxx\all\test_package\build\msvc-193-x86_64-20-release
>       faker-cxx/2.0.0 (test package): Writing generators to C:\J\workspace\prod-v2\bsr\cci-4a790a97\recipes\faker-cxx\all\test_package\build\msvc-193-x86_64-20-release\generators
>       faker-cxx/2.0.0 (test package): Generator 'CMakeDeps' calling 'generate()'
>       faker-cxx/2.0.0 (test package): CMakeDeps necessary find_package() and targets for your CMakeLists.txt
>           find_package(faker-cxx)
>           target_link_libraries(... faker-cxx::faker-cxx)
>       faker-cxx/2.0.0 (test package): Generator 'CMakeToolchain' calling 'generate()'
>       faker-cxx/2.0.0 (test package): CMakeToolchain generated: conan_toolchain.cmake
>       faker-cxx/2.0.0 (test package): CMakeToolchain generated: C:\J\workspace\prod-v2\bsr\cci-4a790a97\recipes\faker-cxx\all\test_package\build\msvc-193-x86_64-20-release\generators\CMakePresets.json
>       faker-cxx/2.0.0 (test package): CMakeToolchain generated: C:\J\workspace\prod-v2\bsr\cci-4a790a97\recipes\faker-cxx\all\test_package\CMakeUserPresets.json
>       faker-cxx/2.0.0 (test package): Generator 'VirtualRunEnv' calling 'generate()'
>       faker-cxx/2.0.0 (test package): Generating aggregated env files
>       faker-cxx/2.0.0 (test package): Generated aggregated env files: ['conanrun.bat', 'conanbuild.bat']
>       
>       ======== Testing the package: Building ========
>       faker-cxx/2.0.0 (test package): Calling build()
>       faker-cxx/2.0.0 (test package): Running CMake.configure()
>       faker-cxx/2.0.0 (test package): RUN: cmake -G "Visual Studio 17 2022" -DCMAKE_TOOLCHAIN_FILE="generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="C:/J/workspace/prod-v2/bsr/cci-4a790a97/recipes/faker-cxx/all/test_package" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" "C:/J/workspace/prod-v2/bsr/cci-4a790a97/recipes/faker-cxx/all/test_package"
>       -- Using Conan toolchain: C:/J/workspace/prod-v2/bsr/cci-4a790a97/recipes/faker-cxx/all/test_package/build/msvc-193-x86_64-20-release/generators/conan_toolchain.cmake
>       -- Conan toolchain: CMAKE_GENERATOR_TOOLSET=v143
>       -- Conan toolchain: C++ Standard 20 with extensions OFF
>       -- The CXX compiler identification is MSVC 19.36.32532.0
>       -- Detecting CXX compiler ABI info
>       -- Detecting CXX compiler ABI info - done
>       -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
>       -- Detecting CXX compile features
>       -- Detecting CXX compile features - done
>       -- Conan: Target declared 'faker-cxx::faker-cxx'
>       CMake Error at build/msvc-193-x86_64-20-release/generators/cmakedeps_macros.cmake:67 (message):
>         Library 'faker-cxx' not found in package.  If 'faker-cxx' is a system
>         library, declare it with 'cpp_info.system_libs' property
>       Call Stack (most recent call first):
>         build/msvc-193-x86_64-20-release/generators/faker-cxx-Target-release.cmake:23 (conan_package_library_targets)
>         build/msvc-193-x86_64-20-release/generators/faker-cxxTargets.cmake:24 (include)
>         build/msvc-193-x86_64-20-release/generators/faker-cxx-config.cmake:16 (include)
>         CMakeLists.txt:4 (find_package)
>       
>       
>       -- Configuring incomplete, errors occurred!
>       See also "C:/J/workspace/prod-v2/bsr/cci-4a790a97/recipes/faker-cxx/all/test_package/build/msvc-193-x86_64-20-release/CMakeFiles/CMakeOutput.log".
>       
>       ERROR: faker-cxx/2.0.0 (test package): Error in build() method, line 20
>       	cmake.configure()
>       	ConanException: Error 1 while executing
>       ```
> 
> 
> **Note:** To save resources, CI tries to finish as soon as an error is found. For this reason you might find that not all the references have been launched or not all the configurations for a given reference. Also, take into account that we cannot guarantee the order of execution as it depends on CI workload and workers availability.

@toge The faker-cxx is not prepared to build with shared + Windows. It can be dropped. I'll check with the upstream.

File: recipes/faker-cxx/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
```
Comment (uilianries) [inline recipes/faker-cxx/all/conanfile.py:67]:
```suggestion
            )
        if self.settings.os == "Windows" and self.options.shared:
            # https://github.com/cieslarmichal/faker-cxx/issues/753
            raise ConanInvalidConfiguration(f"{self.ref} is not prepared to generated shared library on Windows.")
```

Comment (toge) [reply recipes/faker-cxx/all/conanfile.py:67]: @uilianries 
Thank you for your investigation!

**Comment by uilianries** [comment]
> Any progress on the project? @uilianries

@meftunca The PR is ready now. The problem was in the upstream, as it required tons of RAM to be built using GCC. For instance, in my Linux, it stopped and took frozen after consuming 32 GB of RAM. Fortunately, the author improved the project a lot and now does not require that huge amount of memory.

**Review by RubenRBS — APPROVED** [review]
<3 - great follow up @uilianries :)

**Pull request merged**